### PR TITLE
Fix if116 validation

### PR DESCRIPTION
**Pull request by YuvHayun**
## Related Issues
fixes: [CIAC-6649](https://jira-hq.paloaltonetworks.local/browse/CIAC-6649)

## Description
* Added a separated handling for single select and multi select types for incident fields and indicators in **does_not_have_empty_select_values** validation.

### CODE REVIEW

File: demisto_sdk/commands/common/errors.py [hunk 1]
```diff
@@ -1687,8 +1691,13 @@ def field_version_is_not_correct(
 
     @staticmethod
     @error_code_decorator
-    def select_values_cannot_contain_empty_values():
-        return "the field selectValues cannot contain empty values. Please remove."
+    def select_values_cannot_contain_empty_values_in_multi_select_types():
+        return "Multiselect types cannot contain empty values in the field selectValues. Please remove."
```
Comment (dorschw) [inline demisto_sdk/commands/common/errors.py:1695]:
```suggestion
        return "Multiselect types cannot contain empty values in the selectValues field"
```

File: demisto_sdk/commands/common/errors.py [hunk 2]
```diff
@@ -1687,8 +1691,13 @@ def field_version_is_not_correct(
...
+
+    @staticmethod
+    @error_code_decorator
+    def select_values_cannot_contain_multiple_or_only_empty_values_in_single_select_types():
+        return "singleSelect types cannot contain only empty value or more than one empty value in the field selectValues. Please remove."
```
Comment (dorschw) [inline demisto_sdk/commands/common/errors.py:1700]: Please consult with Shirley on the phrasing

File: demisto_sdk/commands/common/tests/field_validator_test.py [hunk 3]
```diff
@@ -475,59 +475,337 @@ def test_is_valid_from_version_field(
...
+        - Case 1: Should return False.
+        - Case 2: Should return False.
+        - Case 3: Should return True.
+        - Case 4: Should return True.
+        - Case 5: Should return True.
+        - Case 6: Should return False.
+        - Case 7: Should return True.
+        - Case 8: Should return False.
+        - Case 9: Should return True.
+        - Case 10: Should return True.
```
Comment (dorschw) [inline demisto_sdk/commands/common/tests/field_validator_test.py:800]:
```suggestion
```
This is already written in the test args

File: demisto_sdk/commands/common/tests/field_validator_test.py [hunk 4]
```diff
@@ -475,59 +475,337 @@ def test_is_valid_from_version_field(
...
+            incident_content,
         )
         structure = StructureValidator(incident_field.path)
         validator = FieldBaseValidator(structure, {"some-type"}, set())
-        assert not validator.does_not_have_empty_select_values()
+        assert validator.does_not_have_empty_select_values() is expected_results
 
-    def test_validate_no_empty_selected_values_value_indicator(self, pack):
+    @pytest.mark.parametrize(
+        "indicator_content, expected_results",
```
Comment (dorschw) [inline demisto_sdk/commands/common/tests/field_validator_test.py:645]: that's a lot of cases, consider using pytest.param to give each an `id` value, making it much easier to understand and maintain

File: demisto_sdk/commands/common/tests/field_validator_test.py [hunk 5]
```diff
@@ -475,59 +475,337 @@ def test_is_valid_from_version_field(
...
+        - Case 1: Should return False.
+        - Case 2: Should return False.
+        - Case 3: Should return True.
+        - Case 4: Should return True.
+        - Case 5: Should return True.
+        - Case 6: Should return False.
+        - Case 7: Should return True.
+        - Case 8: Should return False.
+        - Case 9: Should return True.
+        - Case 10: Should return True.
```
Comment (dorschw) [inline demisto_sdk/commands/common/tests/field_validator_test.py:634]:
```suggestion
```

File: demisto_sdk/commands/common/tests/field_validator_test.py [hunk 6]
```diff
@@ -475,59 +475,337 @@ def test_is_valid_from_version_field(
             == expected
         )
 
-    def test_validate_no_empty_selected_values_value_incident(self, pack):
+    @pytest.mark.parametrize(
+        "incident_content, expected_results",
```
Comment (dorschw) [inline demisto_sdk/commands/common/tests/field_validator_test.py:479]: https://github.com/demisto/demisto-sdk/pull/3382/files#r1269308723

File: demisto_sdk/commands/common/tests/field_validator_test.py (hunk 3 above)
Comment (YuvHayun) [reply demisto_sdk/commands/common/tests/field_validator_test.py:800]: WDYM?

File: demisto_sdk/commands/common/hook_validations/field_base_validator.py [hunk 7]
```diff
@@ -464,14 +464,33 @@ def does_not_have_empty_select_values(self) -> bool:
...
-            (
-                error_message,
-                error_code,
-            ) = Errors.select_values_cannot_contain_empty_values()
+        error_message, error_code = "", ""
+        if self.current_file.get("type") == "singleSelect":
+            select_values = self.current_file.get("selectValues") or []
+            empty_string_appearance = sum(
+                [1 if select_value == "" else 0 for select_value in select_values]
+            )
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/field_base_validator.py:472]:
```suggestion
            empty_string_appearance = sum(
                select_value == "" for select_value in select_values
            )
```
https://stackoverflow.com/a/12765840

File: demisto_sdk/commands/common/errors.py [hunk 8]
```diff
@@ -1687,8 +1691,13 @@ def field_version_is_not_correct(
...
     @error_code_decorator
-    def select_values_cannot_contain_empty_values():
-        return "the field selectValues cannot contain empty values. Please remove."
+    def select_values_cannot_contain_empty_values_in_multi_select_types():
+        return "Multiselect types cannot contain empty values in the selectValues field"
+
+    @staticmethod
+    @error_code_decorator
+    def select_values_cannot_contain_multiple_or_only_empty_values_in_single_select_types():
+        return "singleSelect types cannot contain only empty value or more than one empty value in the field selectValues. Please remove."
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/common/errors.py:1700]:
```suggestion
        return "singleSelect types cannot contain only empty values or more than one empty value in the field selectValues. Please remove."
```

File: demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md [hunk 9]
```diff
@@ -43,6 +43,13 @@ Contributions are welcome and appreciated. For more info, visit our [Contributio
 Contributions are welcome and appreciated. For more info, visit our [Contribution Guide](https://xsoar.pan.dev/docs/contributing/contributing).
 ### Pack Contributors:
 
+---
+ - Contributor1
+ - Contributor2
+
+Contributions are welcome and appreciated. For more info, visit our [Contribution Guide](https://xsoar.pan.dev/docs/contributing/contributing).
+### Pack Contributors:
```
Comment (dorschw) [inline demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md:51]: what's this?

Comment (YuvHayun) [reply demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md:51]: removed, no idea how it was added.. probably some test that didn't finish executing

**Review by dorschw — APPROVED** [review]

**Pull request merged**